envoy/http/async_client.h

```diff
@@ -313,7 +313,7 @@ class AsyncClient {
     // Only used if parent_span_ is set.
     std::string child_span_name_{""};
     // Sampling decision for the tracing span. The span is sampled by default.
```

Comment:
Is it still sampled by default?

Reply:
Yep yep. Only in the [ext_authz span](https://github.com/envoyproxy/envoy/blob/4c40c781f29856f19662f194bbcc767bf1631b79/source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc#L285) do we set `sampled_` to `absl::nullopt`, so that the span doesn't get sampled by default and instead keep the parent's sampling decision.

Reply:
But in the constructor it says sets the value iff the value is present in the constructor, "otherwise keep sampling status of the parent" not "otherwise keep sampling by default"  It sounds like the two comments explain different defaults?  Or I'm confused but in that case we should still ideally make it more clear :-)

Reply:
The two comments refer to two different aspects:

The one from the [RequestOptions struct](https://github.com/envoyproxy/envoy/blob/4c40c781f29856f19662f194bbcc767bf1631b79/envoy/http/async_client.h#L315) says that, by default, the `sampled_` field in `RequestOptions` is `optional<true>`.

The one from the [async_client_impl](https://github.com/envoyproxy/envoy/blob/4c40c781f29856f19662f194bbcc767bf1631b79/source/common/http/async_client_impl.cc#L260) refers to how the async client interprets the value of `RequestOption`'s `sampled_` field: if `sampled_` has any value, then it will `Change span's sampling status`; otherwise it will `keep sampling status of the parent`.

Is this clearer?

Reply:
@alyssawilk I agree keeping parent's sampling default is more reasonable. But here I think @carolinestere is try to keep original behavior so keep the flag default to true.

Reply:
Yep, exactly, I kept the `sampled_` default as `true`, just like it was before, to keep the original behavior.

@alyssawilk I will update the comment in async_client, so it's clear that, by default, the span always gets sampled.

Reply:
@alyssawilk Is it better now? 😄